smsdk/smsdk_entities/factory/factoryV1.py

```diff
@@ -0,0 +1,140 @@
+from typing import List
+import json
+from datetime import datetime, timedelta
+
+try:
+    import importlib.resources as pkg_resources
+except ImportError:
+    # Try backported to PY<37 `importlib_resources`.
+    import importlib_resources as pkg_resources
+
+
+import numpy as np
+
+from smsdk.tool_register import SmsdkEntities, smsdkentities
+from smsdk.utils import module_utility, check_kw
+from smsdk import config
+from smsdk.ma_session import MaSession
+
+ENDPOINTS = json.loads(pkg_resources.read_text(config, "api_endpoints.json"))
+
+@smsdkentities.register("factory_v1")
+class Factory(SmsdkEntities, MaSession):
+    # Decorator to register a function as utility
+    # Only the registered utilites would be accessible
+    # to outside world via client.get_data()
+    mod_util = module_utility()
+
+    def __init__(self, session, base_url) -> None:
+        self.session = session
+        self.base_url = base_url
+
+    @mod_util
+    def get_utilities(self, *args, **kwargs) -> List:
+        """
+        Get the list of registered utilites by name
+        """
+        return [*self.mod_util.all]
+
+    @mod_util
+    def get_factories(self, normalize, *args, **kwargs):
+        """
+        Utility function to get the machines
+        from the ma machine API
+        Recommend to use 'enable_pagination':True for larger datasets
+        """
+        url = "{}{}".format(self.base_url, ENDPOINTS["Factory"]["url_v1"])
+        
+        kwargs = self.modify_input_params(**kwargs)
+        records = self._get_records_mongo_v1(url, normalize, **kwargs)
+        # if not isinstance(records, List):
+        #     raise ValueError("Error - {}".format(records))
+        return records
+
+    def modify_input_params(self, **kwargs):
+        v1_params = ["cb", "select", "where", "group_by", "order_by", "limit", "offset", "per_page", "cursor"]
+
+        list_of_operations = [
+            'ne', 'lt', 'lte', 'gt', 'gte', 'not', 'in', 
+            'nin', 'mod', 'all', 'exists', 'exact', 'iexact', 
+            'contains', 'icontains', 'startswith', 'istartswith', 
+            'endswith', 'iendswith', 'match',
+        ]
+        
+        # Special handling for EF type names
+        machine = kwargs.get('machine__source')
+        machine = machine[1:-1] if machine and machine[0] == "'" else machine
```

Comment:
is this condition valid -->` if machine and machine[0] == "'" `
It has single quote?